### PR TITLE
(FACT-2232) Add Aix networking facts

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

File: lib/facts/aix/interfaces.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+module Facts
+  module Aix
+    class Interfaces
+      FACT_NAME = 'interfaces'
+
+      def call_the_resolver
+        fact_value = Facter::Resolvers::Aix::Networking.resolve(:interfaces)
+
+        Facter::ResolvedFact.new(FACT_NAME, fact_value && !fact_value.empty? ? fact_value.keys.sort.join(',') : nil,
```
[the remark this answers is not in the document]
Comment (IrimieBogdan) [reply lib/facts/aix/interfaces.rb:11]: Maybe we can write it as
```ruby
fact_value  = fact_value&.any? ? nil : fact_value.keys.sort.join(',')
```

File: lib/facts/aix/scope6_interfaces.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      def call_the_resolver
+        resolved_facts = []
+        scope6_interfaces = {}
+        interfaces = Facter::Resolvers::Aix::Networking.resolve(:interfaces)
+        primary = Facter::Resolvers::Aix::Networking.resolve(:primary)
+        interfaces&.each do |interface_name, info|
+          scope6_interfaces["scope6_#{interface_name}"] = info[:scope6] if info[:scope6]
+        end
+
+        scope6_interfaces.each { |fact, value| resolved_facts << Facter::ResolvedFact.new(fact, value, :legacy) }
```
Comment (IrimieBogdan) [inline lib/facts/aix/scope6_interfaces.rb:18]: We could write this as
```ruby
        interfaces&.each do |interface_name, info|
          resolved_facts << Facter::ResolvedFact.new("scope6_#{interface_name}",  info[:scope6], :legacy) if info[:scope6]
        end
```

And drop `scope6_interfaces`

File: spec/facter/facts/aix/networking/interfaces_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    before do
+      allow(Facter::Resolvers::Aix::Networking).to receive(:resolve).with(:interfaces).and_return(value)
+    end
+
+    it 'calls Facter::Resolvers::NetworkingLinux' do
+      fact.call_the_resolver
+      expect(Facter::Resolvers::Aix::Networking).to have_received(:resolve).with(:interfaces)
+    end
+
+    it 'returns networking.interfaces fact' do
```
Comment (Filipovici-Andrei) [inline spec/facter/facts/aix/networking/interfaces_spec.rb:30]: Could you add also the test for when the resolver returns nil. This applies to all the fact tests.

File: lib/resolvers/aix/networking_resolver.rb [hunk 4]
```diff
@@ -15,10 +15,13 @@ def post_resolve(fact_name)
           end
 
           def read_netstat(fact_name)
+            @fact_list[:interfaces] = {}
             output = Facter::Core::Execution.execute('netstat -rn', logger: log)
             output = output.split("\n").select { |line| (line =~ /\s\s[0-9]+.[0-9]+.[0-9]+.[0-9]+|\s\s.*:[0-9a-f]+/) }
```
Comment (IrimieBogdan) [inline lib/resolvers/aix/networking_resolver.rb:20]: `split("\n")` can be replaced with `each_line`

File: lib/resolvers/aix/networking_resolver.rb [hunk 5]
```diff
@@ -27,6 +30,47 @@ def get_primary_interface_info(output)
...
+            ip = ip.gsub(/%[0-9]$/, '') # remove mask information if it exists
+            mask_length = mask_length.to_i - 1 unless is_ipv4
+            @fact_list[:interfaces][name] ||= {}
+            @fact_list[:interfaces][name][bind_to_add] ||= []
+            @fact_list[:interfaces][name][bind_to_add] << ::Resolvers::Utils::Networking.build_binding(ip, mask_length)
+          end
+
+          def populate_with_mtu_and_mac!(interfaces)
+            output = Facter::Core::Execution.execute('netstat -in', logger: log)
+            output.split("\n").each do |line|
```
Comment (IrimieBogdan) [inline lib/resolvers/aix/networking_resolver.rb:60]: `split("\n")` can be replaced with `each_line`

File: lib/resolvers/aix/networking_resolver.rb [hunk 6]
```diff
@@ -15,10 +15,13 @@ def post_resolve(fact_name)
           end
 
           def read_netstat(fact_name)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/resolvers/aix/networking_resolver.rb:17]: Done

File: lib/resolvers/aix/networking_resolver.rb [hunk 7]
```diff
@@ -27,6 +30,47 @@ def get_primary_interface_info(output)
...
+          def extract_interfaces(netstat_output)
+            netstat_output.each do |line|
+              next if line =~ /default/
+
+              info = line.split("\s")
+              mask_length = info[0].match(%r{/([0-9]+)|%([0-9]+)})
+              next unless mask_length
+
+              build_bindings(info[5], info[1], mask_length[1] || mask_length[2],
+                             info[1] =~ /[0-9]+.[0-9]+.[0-9]+.[0-9]+/)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/resolvers/aix/networking_resolver.rb]: Done

File: lib/facts/aix/interfaces.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/facts/aix/interfaces.rb]: Done

File: lib/facts/aix/processor.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  module Aix
+    class Processor
+      FACT_NAME = 'processor.*'
+      TYPE = :legacy
+
+      def call_the_resolver
+        arr = []
+        processors = Facter::Resolvers::Aix::Processors.resolve(:models)
+
+        (0...processors.count).each do |iterator|
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply lib/facts/aix/processor.rb]: Done

File: lib/facts/aix/scope6_interfaces.rb (hunk 2 above)
Comment (oanatmaria) [reply lib/facts/aix/scope6_interfaces.rb]: Done

File: lib/resolvers/aix/networking_resolver.rb (hunk 4 above)
Comment (oanatmaria) [reply lib/resolvers/aix/networking_resolver.rb]: Done

File: lib/resolvers/aix/networking_resolver.rb (hunk 5 above)
Comment (oanatmaria) [reply lib/resolvers/aix/networking_resolver.rb]: Done

File: spec/facter/facts/aix/networking/mtu_spec.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      fact.call_the_resolver
+      expect(Facter::Resolvers::Aix::Networking).to have_received(:resolve).with(:interfaces)
+    end
+
+    it 'calls Facter::Resolvers::Aix::Networking with primary' do
+      fact.call_the_resolver
+      expect(Facter::Resolvers::Aix::Networking).to have_received(:resolve).with(:primary)
+    end
+
+    it 'return mtu fact' do
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/facter/facts/aix/networking/mtu_spec.rb]: Done

File: spec/facter/facts/aix/networking/interfaces_spec.rb (hunk 3 above)
Comment (oanatmaria) [reply spec/facter/facts/aix/networking/interfaces_spec.rb:30]: Done
